services/provider/server/server.go

```diff
@@ -201,3 +216,56 @@ func newClient() (client.Client, error) {
 
 	return client, nil
 }
+
+func (s *OCSProviderServer) getPublicKey(ctx context.Context, secretName string) (string, error) {
+	pubKeySecret := &corev1.Secret{}
+	err := s.client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: s.namespace}, pubKeySecret)
+	if err != nil {
+		return "", fmt.Errorf("failed to get public key secret %q", secretName)
+	}
+
+	publicKey := string(pubKeySecret.Data["key"])
+	if publicKey == "" {
+		return "", fmt.Errorf("public key is not found inside the secret %q", secretName)
+	}
+
+	return publicKey, nil
+}
+
+func validateTicket(ticket, rawPubKey string) error {
+	block, _ := pem.Decode([]byte(rawPubKey))
+	if block == nil {
+		return fmt.Errorf("invalid PEM block")
+	}
+
+	key, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		return fmt.Errorf("failed to parse public key. %v", err)
+	}
+
+	pubKey := key.(*rsa.PublicKey)
+
+	ticketArr := strings.Split(string(ticket), ".")
+	if len(ticketArr) != 0 {
+		return fmt.Errorf("invalid ticket")
+	}
+
+	message, err := base64.StdEncoding.DecodeString(ticketArr[0])
+	if err != nil {
+		return fmt.Errorf("failed to decode payload. %v", err)
+	}
+	signature, err := base64.StdEncoding.DecodeString(ticketArr[1])
+	if err != nil {
+		return fmt.Errorf("failed to decode sigature. %v", err)
+	}
+
+	hash := sha256.Sum256(message)
```

Comment:
Why SHA256? we should use a much simpler hash as we only care about the randomness not the strength of the hash. SHA256 is very expensive. Can we please more to MD5?

Reply:
using MD5 now.